lib/logstash/inputs/syslog.rb

```diff
@@ -206,20 +246,19 @@ def tcp_receiver(output_queue, socket)
   rescue Errno::EBADF
     # swallow connection closed exceptions to avoid bubling up the tcp_listener & server
     logger.info("connection closed", :client => "#{ip}:#{port}")
-  rescue IOError => ioerror
+  rescue IOError => e
     # swallow connection closed exceptions to avoid bubling up the tcp_listener & server
-    raise unless socket.closed? && ioerror.message.include?("closed")
-    logger.info("connection error: #{ioerror.message}")
+    raise(e) unless socket.closed? && e.message.to_s.include?("closed")
+    logger.info("connection error:", :exception => e.class, :message => e.message)
   ensure
     @tcp_sockets.delete(socket)
     socket.close rescue log_and_squash(:close_tcp_receiver_socket)
   end
 
-  private
   def decode(host, output_queue, data)
     @codec.decode(data) do |event|
       decorate(event)
-      event.set("host", host)
+      event.set(@source_key, host) # do we also want to capture source.port ?
```

Comment:
Is the comment used as note during development?

Reply:
good call, thinking about this I am not even sure capturing `source.ip` makes much sense in context of syslog.
maybe we should rather use meta-data ... we're simply acting as a syslog server here.

Reply:
in Beats input the IP is in metadata, maybe the `source.ip` is the IP of the syslog originating host, LS is just an observer in this context

Reply:
this being syslog - we already capture `host.hostname` (could be an IP) from the message e.g.
`<164>Oct 26 15:19:25 some-host a sample message`
so it would make sense to fill in `host.ip` as the client ip (from the socket) and leave port as it (it's irrelevant), 
wdyt?

Reply:
Thanks Andrea, appreciate the feedback - honestly I am not sure how common that "smt in the middle" setup would be.
Or whether it would matter.

> Supposing that `host.hostname` contains the IP as seen by syslog deamon

Usually this isn't a case, you would have something like `some.host` as hostname instead.
And even when it happens you'll have 2 IPs maybe one internal the other external (still fine).

The purpose of starting LS with the syslog input is to use it as a syslog server, so I would assume taking the client IP should be okay. Keep in mind that the existing legacy behavior still fills in `host`.
Also I was suggested `host.ip` would make sense by ECS. 

Do you still prefer that we rather start with smt like `[@metadata][input][syslog][host][ip]` instead?

Reply:
>The purpose of starting LS with the syslog input is to use it as a syslog server,
In this case we can put the `host.ip` with what we read from remote peer socket.

> plugin also supports setting proxy_protocol => true
Maybe in this case we could put the IP in `observer.ip`

Reply:
> plugin also supports setting proxy_protocol => true

it's the same use-case as with `host.ip` I mentioned that in response to the concern:

> suppose you have something in the middle between the syslog client and LS

a common case having smt in the middle would be haproxy to load-balance, in which case using the `proxy_protocol => true` we still get the proper client IP (we want use the one from the socket).